### DISCUSSION TITLE
How to enable membership and allow users edit content from the frontend?

### DISCUSSION THREAD

**Discussion by hiovi** [apostrophecms/apostrophe#3254] [Q&A]
What would be the best way to sign up users from the front end and allow them to, say for instance, rate an article, or create an order. I would like to use only Apostrophe to keep things simple rather than use a number of third party services. Also I would like to avoid creating a separated single page application.

**Comment by boutell** [comment]
Hi @hiovi,

If you are using apostrophe 2.x there is an `apostrophe-signup` module. You can also use `apostrophe-passport` to achieve single sign-on through other platforms like google.

As for rating an article or creating an order, I think you actually should consider using apostrophe-headless (for 2.x), or the built-in REST API support in 3.x, and build your own front end application. Apostrophe has some cool frontend things like the `apostrophe-forms` module that can help with all this, but it's not meant to replace frameworks like React and Vue on the front end.

**Reply by boutell** [reply]
You can load your front end application as part of every page load if you push the assets to an Apostrophe module. This is a common strategy for integrating Webpack with Apostrophe.
